### PR TITLE
[MBL-15820][Teacher] Comment library tests

### PR DESCRIPTION
**Pull request by tamaskozmer**
refs: MBL-15820
affects: Teacher
release note: none

### CODE REVIEW

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/CommentLibraryPageTest.kt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.instructure.teacher.ui
+
+import androidx.test.espresso.Espresso
+import com.instructure.canvas.espresso.mockCanvas.*
+import com.instructure.canvasapi2.models.*
```
Comment (kdeakinstructure) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/CommentLibraryPageTest.kt:21]: Just a hint: You may could set the threshold of using * of particular class imports to 1000 or something big, thus we could see more accurate that what imports are really used in the class.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/CommentLibraryPage.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import com.instructure.teacher.R
+import com.instructure.espresso.page.BasePage
+import com.instructure.espresso.page.onViewWithContentDescription
+import com.instructure.espresso.page.onViewWithText
+
+class CommentLibraryPage : BasePage(R.id.commentLibraryRoot) {
+
+    private val toolbar by OnViewWithId(R.id.commentLibraryToolbar)
+    private val recyclerView by OnViewWithId(R.id.commentLibraryRecyclerView)
+    private val emptyView by OnViewWithId(R.id.commentLibraryEmtpyView, autoAssert = false)
```
Comment (kdeakinstructure) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/CommentLibraryPage.kt:29]: commentLibraryEmptyView

File: automation/espresso/build.gradle [hunk 3]
```diff
@@ -153,6 +155,12 @@ dependencies {
     /* Mock web server */
     implementation("com.squareup.okhttp3:mockwebserver:${Versions.OKHTTP}")
 
+    /* DI */
+    implementation Libs.HILT
+    kapt Libs.HILT_COMPILER
```
Comment (kdeakinstructure) [inline automation/espresso/build.gradle:160]: I'd be curious that what is this exactly. May you can explain a bit in Slack please?

File: automation/espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/MockCanvas.kt [hunk 4]
```diff
@@ -226,6 +226,8 @@ class MockCanvas {
     /** Map of courseId to the courses latest announcement */
     val latestAnnouncements = mutableMapOf<Long, DiscussionTopicHeader>()
 
+    val commentLibraryItems = mutableMapOf<Long, List<String>>()
```
Comment (kdeakinstructure) [inline automation/espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/MockCanvas.kt:229]: I've already wrote above that I'd use the same word for the comments here as well, so i'd rename "Items" to "Suggestions"

File: automation/espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/di/TestGraphQlApiModule.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import com.instructure.canvasapi2.di.GraphQlApiModule
+import com.instructure.canvasapi2.managers.CommentLibraryManager
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.components.SingletonComponent
+import dagger.hilt.testing.TestInstallIn
+
+@Module
+@TestInstallIn(components = [SingletonComponent::class], replaces = [GraphQlApiModule::class])
+class TestGraphQlApiModule {
```
Comment (kdeakinstructure) [inline automation/espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/di/TestGraphQlApiModule.kt:32]: So is it intentional that u want to push this now into master?

File: automation/espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/fakes/FakeCommentLibraryManager.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.instructure.canvas.espresso.mockCanvas.fakes
+
+import com.instructure.canvas.espresso.mockCanvas.MockCanvas
+import com.instructure.canvasapi2.CommentLibraryQuery
+import com.instructure.canvasapi2.managers.CommentLibraryManager
+
+class FakeCommentLibraryManager() : CommentLibraryManager {
```
Comment (kdeakinstructure) [inline automation/espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/fakes/FakeCommentLibraryManager.kt:23]: Cool! :)

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/CommentLibraryPageTest.kt [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+        val filteredSuggestion = "You are off to a great start. Please add more detail to justify your reasoning."
+        commentLibraryPage.assertSuggestionVisible(filteredSuggestion)
+    }
+
+    @Test
+    @TestMetaData(Priority.P2, FeatureCategory.SPEED_GRADER, TestCategory.INTERACTION)
+    fun showEmptyViewWhenFilteringHasNoSuggestion() {
+        goToSpeedGraderCommentsPage()
+        speedGraderCommentsPage.typeComment("great work, bro")
```
Comment (kdeakinstructure) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/CommentLibraryPageTest.kt:123]: I would say the same for this PR :D

Comment (kdeakinstructure) [reply apps/teacher/src/androidTest/java/com/instructure/teacher/ui/CommentLibraryPageTest.kt:123]: Just please put a comment at the end of this line that this comment's purpose is to not filter anything in comment library suggestions. :D

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/CommentLibraryPageTest.kt [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        commentLibraryPage.assertSuggestionVisible(filteredSuggestion)
+        commentLibraryPage.selectSuggestion(filteredSuggestion)
+
+        speedGraderCommentsPage.assertCommentFieldHasText(filteredSuggestion)
+        speedGraderPage.assertCommentLibraryNotVisible()
+    }
+
+    @Test
+    @TestMetaData(Priority.P1, FeatureCategory.SPEED_GRADER, TestCategory.INTERACTION)
+    fun selectCommentLibrarySuggestionAndSendComment() {
```
Comment (kdeakinstructure) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/CommentLibraryPageTest.kt:66]: I think this method could be merged with the previous one, something like this:

commentLibraryPage.selectSuggestion(filteredSuggestion)
speedGraderCommentsPage.assertCommentFieldHasText(filteredSuggestion)
speedGraderPage.assertCommentLibraryNotVisible()

speedGraderCommentsPage.sendComment()
speedGraderCommentsPage.assertDisplaysCommentText(filteredSuggestion)
speedGraderPage.assertCommentLibraryNotVisible()

**Review by kdeakinstructure — APPROVED** [review]

**Review by hermannakos — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**